tests/sqllogictests/suites/base/09_fuse_engine/09_0008_fuse_optimize_table

```diff
@@ -557,25 +557,36 @@ query I
 select a from t10
 ----
 -5
-3
+4
 2
 
 statement ok
-insert into t10 values(1),(2),(3)
+insert into t10 values(1),(0),(3)
 
 statement ok
 optimize table t10 compact
 
 query II
-select segment_count, block_count from fuse_snapshot('db_09_0008', 't10') limit 2
+select segment_count, block_count from fuse_snapshot('db_09_0008', 't10')
 ----
 1 2
+1 2
 2 2
+1 1
+1 1
 
 query TIIFFT
 select * from clustering_information('db_09_0008','t10')
 ----
-(abs(a)) 2 0 1.0 2.0 {"00002":2}
+(abs(a)) 2 0 0.0 1.0 {"00001":2}
+
+statement ok
+optimize table t10 purge limit 2
```

Comment:
During the test, it appears that we do not verify whether the recluster is expected to occur after `compact limit N`.